src/consensus/aft/raft.h

```diff
@@ -141,8 +131,10 @@ namespace aft
     size_t sig_tx_interval;
 
     // Configurations
-    std::list<Configuration> configurations;
-    std::unordered_map<ccf::NodeId, NodeState> nodes;
+    aft::ConfigurationTracker configuration_tracker;
+    std::unordered_map<ccf::NodeId, aft::NodeState> nodes;
+    bool catching_up = false;
```

Comment:
Should this be an enum? I assume you cannot be both catching up and retiring?

Reply:
+1 for rolling this into the state enum unless there's an explicit reason why not.